### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by zmerrychristmas**
Crud with member object and admin

### CODE REVIEW

File: app/Http/Controllers/MemberController.php [hunk 1]
```diff
@@ -12,10 +12,29 @@ class MemberController extends Controller
      *
```
Comment (4c0n) [inline app/Http/Controllers/MemberController.php:13]: Please add the `.phpintel` folder to your `.gitignore` file. You don't need these files in your repository.

File: app/Http/Controllers/MemberController.php [hunk 2]
```diff
@@ -12,10 +12,29 @@ class MemberController extends Controller
...
      * @return \Illuminate\Http\Response
      */
-    public function index()
+    public function index(Request $request)
     {
         $members = Member::all();
-        return response()->json($members);
+        $messages = [];
+        $action = $request->get('ACTION');
```
Comment (4c0n) [inline app/Http/Controllers/MemberController.php]: I think you should do this following the REST standard (using http methods):
Create = POST
Read = GET
Update = PUT/PATCH
Delete = DELETE

You can specify 1 method for create, 1 method for update, etc.. And bind them to the method and uri in your routes.php file

File: app/Http/Controllers/MemberController.php [hunk 3]
```diff
@@ -76,10 +104,30 @@ public function update(Request $request, $id)
         $member = Member::find($id);
         $member->name = $request->get('name');
         $member->information = $request->get('information');
-        $member->date_of_birth = $request->get('date_of_birth');
+        $member->date_of_birth = date("Y-m-d H:i:s", strtotime($request->get('dob')));
         $member->position = $request->get('position');
         $member->phone = $request->get('phone');
         $member->gender = $request->get('gender');
+        if ($request->hasFile('avatar') && $request->file('avatar')->isValid()) {
```
Comment (4c0n) [inline app/Http/Controllers/MemberController.php:118]: Good you're deleting the old one! :D

File: app/Http/Controllers/MemberController.php [hunk 4]
```diff
@@ -101,6 +149,15 @@ public function destroy($id)
...
+        return view('member/index', ['title' => 'Members']);
+    }
+
+    public function newMember()
+    {
+        return view('member/index', ['title' => 'Create  Member']);
+    }
+    public function editMember()
+    {
+        return view('member/index', ['title' => 'Edit  Member']);
```
Comment (4c0n) [inline app/Http/Controllers/MemberController.php:168]: You don't need these functions

File: app/Http/Controllers/ProjectController.php [hunk 5]
```diff
@@ -117,4 +117,19 @@ public function assignMember(Request $request)
...
+
+    public function newProject()
+    {
+        return view('project/index', ['title' => 'Create  Project']);
+    }
+
+    public function editProject()
+    {
+        return view('project/index', ['title' => 'Edit  Project']);
+    }
```
Comment (4c0n) [inline app/Http/Controllers/ProjectController.php:157]: Don't need these functions

File: resources/assets/js/components/UpdateProject.js [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount(){
+    axios.get(MyGlobleSetting.url + `/api/projects/${this.props.params.id}`)
+    .then(response => {
+      this.setState({ projectName: response.data.name, projectInformation: response.data.information, projectPhone: response.data.phone, projectDob: response.data.date_of_birth, projectPosition: response.data.position, projectGender: response.data.gender, projectAvatar: response.data.avatar });
+    })
+    .catch(function (error) {
+      console.log(error);
```
Comment (4c0n) [inline resources/assets/js/components/UpdateProject.js]: Maybe it is better to send this error message to the error console: `console.error(error)`

**Review by 4c0n — CHANGES_REQUESTED** [review]
Thanks, it seems you are making progress. Don't forget to add the test code for the backend.

File: app/Http/Controllers/MemberController.php (hunk 1 above)
Comment (zmerrychristmas) [reply app/Http/Controllers/MemberController.php:13]: need update git for it, i adrealy at this line to .gitignore

File: app/Http/Controllers/MemberController.php (hunk 4 above)
Comment (zmerrychristmas) [reply app/Http/Controllers/MemberController.php:168]: fix

File: app/Http/Controllers/ProjectController.php (hunk 5 above)
Comment (zmerrychristmas) [reply app/Http/Controllers/ProjectController.php:157]: fix

**Comment by zmerrychristmas** [comment]
fixxxed

**Pull request merged**